org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageClientImpl.java

```diff
@@ -83,6 +94,51 @@ public final void logMessage(MessageParams message) {
 		CompletableFuture.runAsync(() -> ServerMessageHandler.logMessage(wrapper, message));
 	}
 
+	private Map<String, BlockingQueue<ProgressParams>> progressQueueMap = new HashMap<>();
+
+	@Override
+	public CompletableFuture<Void> createProgress(WorkDoneProgressCreateParams params) {
+		String jobName = params.getToken().getLeft();
```

Comment:
Using the `token` as `jobName` may cause some jobs to be having a weird UX (e.g. our Language Server uses UUIDs as progress tokens). Maybe put at least the Language Server name in front of the token or use the message field in `WorkDoneProgressCreateParams`?

Reply:
done